content/vendor/snapshots/snapshots.md

```diff
@@ -0,0 +1,164 @@
+
+### Velero
+
+The customer must have [velero](https://velero.io/) installed in the `velero` namespace along with [Restic](https://velero.io/docs/v1.2.0/restic/).
+
+#### Modify default install (this is going away)
+
+The following patches must be applied to the default velero install with the CLI.
+
+Load all provider plugins as initContainers. (Velero deployment only, not applicable to restic daemonset).
+
+[velero deployment](https://github.com/replicatedhq/kURL/blob/master/addons/velero/1.2.0/deployment.yaml#L20-L34)
+
+Mount secrets volumes in subdirectory
+
+[velero deployment](https://github.com/replicatedhq/kURL/blob/master/addons/velero/1.2.0/deployment.yaml#L48-L53)
+
+[restic daemonset](https://github.com/replicatedhq/kURL/blob/master/addons/velero/1.2.0/restic-daemonset.yaml#L20-L31)
+
+Use env vars to point to provider-specific credentials
+
+[velero deployment](https://github.com/replicatedhq/kURL/blob/master/addons/velero/1.2.0/deployment.yaml#L64-L69)
+
+[restic daemonset](https://github.com/replicatedhq/kURL/blob/master/addons/velero/1.2.0/restic-daemonset.yaml#L46-L51)
+
+Use provider-specific secret volumes
+
+[velero deployment](https://github.com/replicatedhq/kURL/blob/master/addons/velero/1.2.0/deployment.yaml#L71-L82)
+
+[restic daemonset](https://github.com/replicatedhq/kURL/blob/master/addons/velero/1.2.0/restic-daemonset.yaml#L68-L73)
+
+
+## Release YAML Configuration
+
+### Backup
+
+Kotsadm requires exactly one [velero.io/v1/Backup](https://velero.io/docs/v1.2.0/api-types/backup/) resource be included in the release yaml.
+Templates are supported in this file.
+All fields in the metadata are irrelevant.
+This yaml is never applied directly to Kubernetes - it is saved until a backup is triggered.
+After rendering it, kotsadm will pick the `includedNamespaces` and `hooks` fields from the spec.
+No other fields are added to the Backup kotsadm generates and applies.
+
+`spec.includedNamesapces` tells Velero which namespaces should be included in the snapshot.
+If this is not set in the Backup from the release yaml, kotsadm will add its own namespace.
+If the application has hardcocded namespaces into its yaml configs, all those namespaces should be specified in this list.
+
+All objects in included namespaces will be snapshotted, including objects that are not found in release yaml and are dynamically created by other components or manually created by users with kubectl.
+There are two exceptions:
+
+1. If the Backup includes the same namespace that kotsadm is running in, then kotsadm will add a label selector to the Backup spec to exclude all kotsadm objects.
+2. Invidual resources in your release can be excluded from the backup by adding the `velero.io/exclude-from-backup=true` label to them.
+
+#### Example
+
+Below is an example of a Backup that hardcodes a namespace and adds some pre-exec hooks.
+
```

Comment:
This example helps. Maybe it may help to start with this. After I read this, I more easily understand a lot of what is mentioned above.

Reply:
moved